### PR TITLE
feature: Add setup.py to allow installation

### PR DESCRIPTION
**Pull request by maartenarnst**
This PR adds a `setup.py` to try to allow installation as a Python package.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,3 +25,12 @@ jobs:
...
                     PYTHONPATH=$PWD pytest tests/test_topology.py
+
+    install-as-package:
+        runs-on: [ubuntu-latest]
+        container:
+            image: python:3.9
+        steps:
+            - name: Install as package.
+              run : |
+                pip3 install git+https://github.com/uliegecsm/hwloc-xml-parser.git@make_package
```
Comment (romintomasetti) [inline .github/workflows/test.yml:36]: To make the test strong, you should:
- install (as done right now)
- use it

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from setuptools import setup, find_packages
+
+setup(
+    name='hwloc-xml-parser',
+    version='0.1',
```
Comment (romintomasetti) [inline setup.py:5]: align

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -19,9 +19,28 @@ jobs:
 
             - name: Prepare.
               run : |
-                    pip install -r requirements.txt
-                    pip install pytest
+                pip install -r requirements.txt
+                pip install pytest
```
Comment (romintomasetti) [inline .github/workflows/test.yml:23]: We use 4 spaces indentation :wink:

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -19,9 +19,28 @@ jobs:
...
+            image: python:3.9
+        steps:
+            - name: Prepare.
+              run : |
+                  apt update
+                  apt install -y hwloc
+
+            - name: Install as package.
+              run : |
+                  pip3 install git+https://github.com/uliegecsm/hwloc-xml-parser.git@make_package
```
Comment (romintomasetti) [inline .github/workflows/test.yml:42]:
```suggestion
                  pip3 install git+https://github.com/uliegecsm/hwloc-xml-parser.git@${{ github.sha }}
```
(to make it bullet proof)

File: hwloc_xml_parser/tests/data/intel-core-i7-4790.xml
Comment (romintomasetti) [inline hwloc_xml_parser/tests/data/intel-core-i7-4790.xml:1]: revert

File: hwloc_xml_parser/tests/test_topology.py
Comment (romintomasetti) [inline hwloc_xml_parser/tests/test_topology.py:1]: revert

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -19,9 +19,27 @@ jobs:
...
               run : |
-                    pip install -r requirements.txt
-                    pip install pytest
+                  pip install -r requirements.txt
+                  pip install pytest
 
             - name: Run tests.
               run : |
-                    PYTHONPATH=$PWD pytest tests/test_topology.py
+                  PYTHONPATH=$PWD/hwloc_xml_parser pytest tests/test_topology.py
```
Comment (romintomasetti) [inline .github/workflows/test.yml:27]:
```suggestion
                  PYTHONPATH=$PWD pytest tests/test_topology.py
```

and write `from hwloc_xml_parser.topology import SystemTopology` in the test ?

File: .github/workflows/test.yml (hunk 4 above)
Comment (maartenarnst) [reply .github/workflows/test.yml]: OK. Thanks. Better indeed!

**Comment by maartenarnst** [comment]
Hi @romintomasetti, this one is ready for a thumps up!

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -19,9 +19,27 @@ jobs:
...
+
+    install-as-package:
+        runs-on: [ubuntu-latest]
+        container:
+            image: python:3.9
+        steps:
+            - name: Prepare.
+              run : |
+                  apt update
+                  apt install -y hwloc
```
Comment (romintomasetti) [inline .github/workflows/test.yml:37]:
```suggestion
                  apt --yes --no-install-recommends install hwloc
```

Comment (maartenarnst) [reply .github/workflows/test.yml]: OK. Will take less time indeed!

**Review by romintomasetti — APPROVED** [review]

**Pull request merged**